charts/karpenter/templates/controller/deployment.yaml

```diff
@@ -15,20 +17,24 @@ kind: Deployment
 metadata:
   name: karpenter-controller
   namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "karpenter.labels" . | indent 4 }}
 spec:
   replicas: {{ .Values.controller.replicas }}
   strategy:
     type: Recreate
   selector:
     matchLabels:
       karpenter: controller
+      {{- include "karpenter.selectorLabels" . | indent 6 }}
   template:
     metadata:
       labels:
         karpenter: controller
+        {{- include "karpenter.labels" . | indent 8 }}
```

Comment:
Shouldn't this match line 29?

Reply:
@ellistarn line 34 include all labels, instead the line 29 only labels used for selection. I made a mistake ?